src/target/target.c

```diff
@@ -3046,44 +3046,41 @@ static int handle_target(void *priv)
 			is_jtag_poll_safe() && target;
 			target = target->next) {
 
-		if (!target->tap->enabled)
+		/* This function only gets called every polling_interval, so
+		 * allow some slack in the time comparison. Otherwise, if we
+		 * schedule for now+polling_interval, the next poll won't
+		 * actually happen until a polling_interval later. */
+		if (!target->tap->enabled ||
+		    power_dropout ||
+		    srst_asserted ||
+		    timeval_ms() + polling_interval / 2 < target->backoff.next_attempt)
 			continue;
 
-		if (target->backoff.times > target->backoff.count) {
-			/* do not poll this time as we failed previously */
-			target->backoff.count++;
-			continue;
+		/* polling may fail silently until the target has been examined */
+		retval = target_poll(target);
+		if (retval == ERROR_OK) {
+			target->backoff.interval = polling_interval;
+		} else {
+			/* Increase interval between polling up to 5000ms */
+			target->backoff.interval = MAX(polling_interval,
+					MIN(target->backoff.interval * 2 + 1, 5000));
+			/* Tell GDB to halt the debugger. This allows the user to run
+			* monitor commands to handle the situation. */
```

Comment:
Cosmetic detail - indent. 

```suggestion
			/* Tell GDB to halt the debugger. This allows the user to run
			 * monitor commands to handle the situation. */
```